Tech@HubAPI/Tech@HubAPI/Models/User.cs

```diff
@@ -28,7 +30,19 @@ public User(string username, byte[] password, byte[] salt, string email, string
 
         public byte[] Salt { get; set; }
 
-        public string Email { get; set; }
+        public string Email {
+            get
+            {
+                return this.Email;
+            }
+            set
+            {
+                if (!Email.Contains('@') || !Email.Contains('.') || Email.IndexOf('@') > Email.IndexOf('.'))
```

Comment:
It could be risky implementing this check on the user itself. It could cause an exception to be thrown in a place we don't expect. Instead, can you implement a function on the `SignupForm` class called `void Validate()`? This function will do the exact same check here and throw the same exception if the email is invalid. Then just call `form.Validate()` back in the endpoint and if that throws an exception return the `BadRequest()`

Reply:
The `Validate()` function can also attempt to convert the `form.BirthDate` field to a `DateTime` object and check if that is valid